playbook/app/pb_kits/playbook/pb_currency/_currency.jsx

```diff
@@ -69,6 +71,20 @@ const Currency = (props: CurrencyProps) => {
     className
   )
 
+  const getAbbreviatedValue = (arg) => {
+    const num = new Intl.NumberFormat('en-US', {
+      notation: 'compact',
+      maximumFractionDigits: 1,
+    }).format(whole.split(',').join('')).toString()
+    let output
+    if (arg === 'amount') {
+      output = num.slice(0, -1)
+    } else if (arg === 'unit') {
+      output = num.slice(-1)
+    }
+    return output
+  }
```

Comment:
```suggestion
  const getFormattedNumber = (input) => new Intl.NumberFormat('en-US', {
      notation: 'compact',
      maximumFractionDigits: 1,
    }).format(input)
  
  type AbbrType = 'amount' | 'unit'
  
  const getAbbreviatedValue = (abbrType: AbbrType) => {
    const num = `${getFormattedNumber(whole.split(',').join(''))}`,
          isAmount = abbrType === 'amount',
          isUnit =  abbrType === 'unit'
               
    return isAmount ? num.slice(0, -1) : isUnit ? num.slice(-1) : ''
  }
```

A couple of things about the original code here:

- Lower readability 
- The number formatter should be abstracted, especially to allow future re-use. Doing so also helps readability
  - Functions should strive to do one thing well and preferably only one
- `arg` is way to generic for *any* use case. Name your arguments contextually and don't forget to specify types. Now we can tell exactly the values that can be accepted. 
- The function was not returning any value if neither enumerated value was provided. It would just be `undefined` which would be bad